src/deepstream.io.js

```diff
@@ -121,44 +155,23 @@ Deepstream.prototype.isRunning = function () {
  * @returns {void}
  */
 Deepstream.prototype.start = function () {
-  if (this._currentState !== STATES.CLOSED) {
-    throw new Error(`Server can only start after it stops succesfully, currently ${this._currentState}`)
+  if (!this._state.is(STATES.STOPPED)) {
+    throw new Error(`Server can only start after it stops successfully. Current state: ${this._state.state}`)
   }
-  this._currentState = STATES.STARTING
   this._showStartLogo()
-  const loggerInitializer = new DependencyInitialiser(this._options, 'logger')
-  loggerInitializer.once('ready', this._start.bind(this))
-}
-
-/**
- * This is the actual function which starts deepstream. It is invoked after+
- * the logger was intialized or emitted the read event if it was initialized
- * asynchronously
- *
- * @private
- * @returns {void}
- */
-Deepstream.prototype._start = function () {
-  this._options.logger.log(C.LOG_LEVEL.INFO, C.EVENT.INFO, `deepstream version: ${pkg.version}`)
-
-  // otherwise (no configFile) deepstream was invoked by API
-  if (this._configFile != null) {
-    this._options.logger.log(C.LOG_LEVEL.INFO, C.EVENT.INFO, `configuration file loaded from ${this._configFile}`)
-  }
-
-  if (global.deepstreamLibDir) {
-    this._options.logger.log(C.LOG_LEVEL.INFO, C.EVENT.INFO, `library directory set to: ${global.deepstreamLibDir}`)
-  }
-
-  let i
-  let initialiser
-
-  this._options.pluginTypes.forEach((pluginType) => {
-    initialiser = new DependencyInitialiser(this._options, pluginType)
-    initialiser.once('ready', this._checkReady.bind(this, pluginType, initialiser.getDependency()))
+  process.nextTick(() => this._state.start())
+
+  process.on('SIGINT', () => {
```

Comment:
You're right, this made it into the PR by mistake. See the changes to bin/deepstream-start.js